packages/script-editor/src/test/script-editor.spec.ts

```diff
@@ -0,0 +1,98 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { JupyterServer } from '@jupyterlab/testutils';
+
+import { ScriptEditorController } from '../ScriptEditorController';
+import { ScriptRunner } from '../ScriptRunner';
+jest.setTimeout(3 * 60 * 1000);
+
+const server = new JupyterServer();
+const language = 'python';
+
+beforeAll(async () => {
+  await server.start();
+});
+
+afterAll(async () => {
+  await server.shutdown();
+});
+
+describe('@elyra/script-editor', () => {
+  describe('ScriptEditorController', () => {
+    describe('#getKernelSpecs', () => {
+      it('should get Python kernel specs', async () => {
+        const controller = new ScriptEditorController();
+        const kernelSpecs = await controller.getKernelSpecsByLanguage(language);
+        for (const [key, value] of Object.entries(
+          kernelSpecs?.kernelspecs ?? []
+        )) {
+          expect(key).toContain(language);
+          expect(value?.language).toContain(language);
+        }
+      });
+    });
+  });
+
+  describe('KernelManager', () => {
+    describe('#startSession', () => {
+      let runner: ScriptRunner;
+      let kernelName: string;
+      const testPath = 'test.py';
+
+      beforeEach(async () => {
+        runner = new ScriptRunner((x: boolean): void => console.log(x));
+        const controller = new ScriptEditorController();
+        const kernelSpecs = await controller.getKernelSpecsByLanguage(language);
+        kernelName =
+          Object.values(kernelSpecs?.kernelspecs ?? [])[0]?.name || '';
+      });
+
+      it('should start a kernel session', async () => {
+        const session = await runner.startSession(kernelName, testPath);
+        expect(session.id).toEqual(runner.sessionConnection?.id);
+        expect(runner.sessionConnection?.kernel?.connectionStatus).toEqual(
+          'connecting'
+        );
+        runner.shutdownSession();
+      });
+
+      it('should shut down a kernel session', async () => {
+        await runner.startSession(kernelName, 'test.py');
+        await runner.shutdownSession();
+        expect(runner.sessionConnection).toBeNull();
+      });
+
+      it('should run script', async () => {
+        const code = 'print("Test")';
+        const kernelMsgFn = async (msg: any): Promise<void> =>
+          msg.output && expect(msg.output).toMatch(/test/i);
+        expect(kernelName).not.toBe('');
+        await runner.runScript(kernelName, testPath, code, kernelMsgFn);
+        await runner.shutdownSession();
+      });
+
+      it('should receive error message when running a broken script', async () => {
+        const code = 'print(Broken Test)';
+        const kernelMsgFn = async (msg: any): Promise<void> =>
+          msg.error && expect(msg.error.type).toMatch(/error/i);
+        expect(kernelName).not.toBe('');
+        await runner.runScript(kernelName, testPath, code, kernelMsgFn);
```

Comment:
You should check for failure here

Reply:
Same as comment above, the error check for this test is being called inside `kernelMsgFn`